stdlib/public/Concurrency/AsyncSequence.swift

```diff
@@ -12,16 +12,112 @@
+/// over `Counter`, a custom `AsyncSequence` that produces `Int` values from
+/// `1` up to a `howHigh` value:
+///
+///     for await i in Counter(howHigh: 10) {
+///       print(i, terminator: " ")
+///     }
+///     // Prints: 1 2 3 4 5 6 7 8 9 10
+///
+/// An `AsyncSequence` doesn't generate or contain the values; it just defines
+/// how you access them. Along with defining the type of values as an associated
+/// type called `Element`, the `AsyncSequence` defines a `makeAsyncIterator()`
+/// method. This returns an instance of type `AsyncIterator`. Like the standard
+/// `IteratorProtocol`, the `AsyncIteratorProtocol` defines a single `next()`
+/// method to produce elements. The difference is that the `AsyncIterator`
+/// defines its `next()` method as `async`, which requires a caller to wait for
+/// the next value with the `await` keyword.
+///
+/// `AsyncSequence` also defines functions for processing the elements you
+/// receive, modeled on the operations provided by the basic `Sequence` in the
+/// standard library. There are two categories of functions: those that return a
+/// single value, and those that return another `AsyncSequence`.
+///
+/// Single-value functions eliminate the need for a `for await`-`in` loop, and instead
+/// let you make a single `await` call. For example, the `contains(_:)` method
+/// returns a Boolean value that indicates if a given value exists in the
+/// `AsyncSequence`. Given the `Counter` sequence from the previous example,
+/// you can test for the existence of a sequence member with a one-line call:
+///
+///     let found = await Counter(howHigh: 10).contains(5) // true
+///
+/// Functions that return another `AsyncSequence` return a type specific to the
+/// function's semantics. For example, the `.map(_:)` operator returns a
+/// `AsyncMapSequence` (or a `AsyncThrowingMapSequence`, if the closure you
+/// provide to the `map(_:)` function can throw an error). These returned
+/// sequences don't eagerly await the next member of the sequence, which allows
+/// the caller to decide when to start work. Typically, you'll iterate over
+/// these sequences with `for await`-`in`, like the base `AsyncSequence` you started
+/// with. In the following example, the `map(_:)` function transforms each `Int`
+/// received from a `Counter` sequence into a `String`:
+///
+///     let stream = Counter(howHigh: 10)
+///       .map { $0 % 2 == 0 ? "Even" : "Odd" }
+///     for await s in stream {
+///         print(s, terminator: " ")
+///     }
+///     // Prints: Odd Even Odd Even Odd Even Odd Even Odd Even
+///
 @available(SwiftStdlib 5.5, *)
 @rethrows
 public protocol AsyncSequence {
+  /// The type of element produced by this asynchronous sequence.
   associatedtype AsyncIterator: AsyncIteratorProtocol where AsyncIterator.Element == Element
+  /// Creates the asynchronous iterator that produces elements of this
+  /// asynchronous sequence.
```

Comment:
Same question about whether you need a blank line after the abstract.

Reply:
Added.